ambry-store/src/main/java/com.github.ambry.store/Journal.java

```diff
@@ -154,6 +167,23 @@ void addEntry(Offset offset, StoreKey key) {
     return journalEntries;
   }
 
+  /**
+   * @return all journal entries at the time when this method is invoked. (Return empty list if journal is empty)
+   */
+  List<JournalEntry> getAllEntries() {
```

Comment:
i am a little concerned about how often this is going to be invoked. It seems like a huge waste that most of the result would be thrown away.

Reply:
it's not cheap but this method is called rarely, only when replica is being decommissioned.